src/overlays/actors/ovl_Dm_Char08/z_dm_char08.c

```diff
@@ -28,71 +63,1061 @@ const ActorInit Dm_Char08_InitVars = {
+    sLargeTurtleBubbleAccel.z = 0.0f;
+    for (i = 0; i < this->bubbleCount; i++) {
+        EffectSsDtBubble_SpawnCustomColor(play, &this->bubblePos, &velocity, &sLargeTurtleBubbleAccel,
+                                          &sLargeTurtleBubblePrimColor, &sLargeTurtleBubbleEnvColor,
+                                          Rand_S16Offset(80, 130), 30, 0);
+    }
+}
+
+void func_80AAFCCC(DmChar08* this, PlayState* play) {
+    switch (play->sceneId) {
+        case SCENE_31MISAKI:
+            if (!(gSaveContext.save.weekEventReg[55] & 0x80)) {
+                switch (this->unk_206) {
+                    case 0:
+                        break;
+
+                    case 1:
+                        // Fire hookshot
+                        Message_StartTextbox(play, 0x102A, &this->dyna.actor);
+                        this->unk_206++;
+                        break;
+
+                    case 2:
+                        // I know your name
+                        if (Message_GetState(&play->msgCtx) == TEXT_STATE_CLOSING) {
+                            Message_StartTextbox(play, 0x102B, &this->dyna.actor);
+                            this->unk_206 = 0;
+                        }
+                        break;
+                }
+            } else {
+                switch (this->unk_206) {
+                    case 0:
+                        break;
+
+                    case 1:
+                        // Going to the temple again
+                        Message_StartTextbox(play, 0x103B, &this->dyna.actor);
+                        this->unk_206++;
+                        break;
+
+                    case 2:
+                        if (Message_GetState(&play->msgCtx) == TEXT_STATE_CLOSING) {
+                            func_801477B4(play);
+                            this->unk_206 = 0;
+                        }
+                        break;
+                }
+            }
+            return;
+
+        case SCENE_SEA:
+            switch (this->unk_206) {
+                case 0:
+                    break;
+
+                case 1:
+                    Message_StartTextbox(play, 0x102E, &this->dyna.actor);
+                    this->unk_206++;
+                    break;
+
+                case 2:
+                    if (Message_GetState(&play->msgCtx) == TEXT_STATE_CLOSING) {
+                        Message_StartTextbox(play, 0x102F, &this->dyna.actor);
+                        this->unk_206 = 0;
+                    }
+                    break;
+            }
+            break;
+    }
+}
+
+void DmChar08_DoNothing(DmChar08* this, PlayState* play) {
+}
+
+void func_80AAFE88(DmChar08* this, PlayState* play) {
+    s32 actorActionIndex;
+    CsCmdActorAction* csAction;
+    f32 phi_f12;
+
+    if (Cutscene_CheckActorAction(play, 474)) {
+        actorActionIndex = Cutscene_GetActorActionIndex(play, 474);
+        if (this->unk_1F6 != play->csCtx.actorActions[actorActionIndex]->action) {
```

Comment:
Other places in the code base I've seen this called `csAction`, don't know if the same could apply here.